srtp/srtp.c

```diff
@@ -1532,6 +1532,110 @@ srtp_get_session_keys(srtp_stream_ctx_t *stream, uint8_t* hdr,
   return NULL;
 }
 
+static srtp_err_status_t
+srtp_estimate_index(srtp_rdbx_t *rdbx,
+                    uint32_t roc,
+                    srtp_xtd_seq_num_t *est,
+                    srtp_sequence_number_t seq,
+                    int *delta)
+{
+#ifdef NO_64BIT_MATH
+  uint32_t internal_pkt_idx_reduced;
+  uint32_t external_pkt_idx_reduced;
+  uint32_t internal_roc;
+  uint32_t roc_difference;
+#endif
+
+#ifdef NO_64BIT_MATH
+  *est = (srtp_xtd_seq_num_t)make64(roc >> 16, (roc << 16) | seq);
+  *delta = low32(est) - rdbx->index;
+#else
+  *est = (srtp_xtd_seq_num_t)(((uint64_t)roc) << 16) | seq;
+  *delta = (int)(*est - rdbx->index);
+#endif
+
+  if (*est > rdbx->index) {
+#ifdef NO_64BIT_MATH
+    internal_roc = (uint32_t)(rdbx->index >> 16);
+    roc_difference = roc - internal_roc;
+    if (roc_difference > 1) {
+      *delta = 0;
+      return srtp_err_status_pkt_idx_adv;
+    }
+
+    internal_pkt_idx_reduced = (uint32_t)(rdbx->index & 0xFFFF);
+    external_pkt_idx_reduced = (uint32_t)((roc_difference << 16) | seq);
```

Comment:
@paulej,

<Perhaps I overlooked it, but what happens if you attempt to set the ROC value 
<prior to encrypting a packet? Did I miss the logic that handles that flow?

I did not add it since I had no use case for it my self. 
But it is added now and I will push it tomorrow plus some tests for it

<Why are these "_reduced" variables not 64-bit values? What if I set the ROC to 
<be some much larger number, like 0xFFFFFF00, when the original ROC was 0?

I have to admit that I just took the code from the EKT branch without studying it too much